app/[id]+api.ts

```diff
@@ -1,79 +1,86 @@
-import { deleteGuest, getGuest, Guest } from '../database/guests';
+import { deleteGuestInsecure, getGuestInsecure } from '../database/guests';
+import { ExpoApiResponse } from '../ExpoApiResponse';
+import { Guest, guestsSchema } from '../migrations/00000-createTableGuests';
+
+type ExpoResponseBodyGet =
+  | {
+      guest: Guest;
+    }
+  | { error: string };
+
+export async function GET(
+  request: Request,
+  { id }: { id: string },
+): Promise<ExpoApiResponse<ExpoResponseBodyGet>> {
+  const guest = await getGuestInsecure(Number(id));
 
-export function GET(request: Request, { id }: { id: string }) {
-  const guest = getGuest(Number(id));
-  console.log(guest);
   if (!guest) {
-    return Response.json(
+    return ExpoApiResponse.json(
       { error: `No guest with id ${id} found` },
       { status: 404 },
     );
   }
-  return Response.json(guest);
+  return ExpoApiResponse.json({ guest: guest });
 }
 
-export function DELETE(request: Request, { id }: { id: string }) {
-  console.log('delete guest', id);
+type ExpoResponseBodyDelete =
+  | {
+      guest: Guest;
+    }
+  | { error: string };
 
-  const guests = deleteGuest(Number(id));
+export async function DELETE(
+  request: Request,
+  { id }: { id: string },
+): Promise<ExpoApiResponse<ExpoResponseBodyDelete>> {
+  const guest = await deleteGuestInsecure(Number(id));
+
+  if (!guest) {
+    return ExpoApiResponse.json(
+      { error: `Guest ${id} not found` },
+      { status: 404 },
+    );
+  }
 
-  return Response.json(guests);
+  return ExpoApiResponse.json({ guest: guest });
 }
 
+type ExpoResponseBodyPut =
+  | {
+      guest: Guest;
+    }
+  | { error: string; errorIssues: { message: string }[] }
+  | { error: string };
```

Comment:
what about making `errorIssues` potentially undefined instead? (with the question mark)

I'll add this also to the Next.js example, it's missing there:

https://github.com/upleveled/next-js-example-fall-2024-atvie/blob/af2e2c1b65cd97de2af842fd109b294524acf256/app/api/animals/%5BanimalId%5D/route.ts

Reply:
Changed it to 
```tsx
type GuestResponseBodyPut =
  | {
      guest: Guest;
    }
  | { error: string; errorIssues?: { message: string }[] };
```